mono/metadata/profiler.c

```diff
@@ -1220,7 +1220,6 @@ void
 mono_profiler_load (const char *desc)
 {
 	char *cdesc = NULL;
-	mono_gc_base_init ();
```

Comment:
This change is ok, but one thing needs to be done to ensure the log profiler won't break: The `mono_counters_register` calls [here](https://github.com/mono/mono/blob/master/mono/profiler/proflog.c#L4735) must be moved to [here](https://github.com/mono/mono/blob/master/mono/profiler/proflog.c#L4431) (just after the write to `runtime_inited`).